src/python/pants/backend/python/providers/pyenv/rules.py

```diff
@@ -159,25 +157,30 @@ async def get_pyenv_install_info(
                     dedent(
                         f"""\
                         import fcntl
+                        import os
                         import pathlib
+                        import shutil
                         import subprocess
                         import sys
 
                         PYENV_ROOT = pathlib.Path("{PYENV_NAMED_CACHE}", "{installation_fingerprint}").resolve()
                         SPECIFIC_VERSION = sys.argv[1]
                         SPECIFIC_VERSION_PATH = PYENV_ROOT / "versions" / SPECIFIC_VERSION
-                        DONEFILE_PATH = SPECIFIC_VERSION_PATH / "DONE"
-                        DONEFILE_LOCK_PATH = SPECIFIC_VERSION_PATH / "DONE.lock"
-                        DONEFILE_LOCK_FD = DONEFILE_LOCK_PATH.open(mode="w")
+                        DONEFILE_DIR = PYENV_ROOT / "done" / SPECIFIC_VERSION
```

Comment:
Idk why but it's weird to me to have a dedicated "done" directory.

Would it work to have a file at the root called `<specific version>.DONE`?
 Same for the lockfile?

Reply:
I dunno, it felt better to me to sequester these under a directory that is easy to clean out or inspect, rather than mingle with pyenv's own files. I have no idea how pyenv would react to seeing version-looking files in that dir, seems like an invitation for trouble.

Reply:
Ok let's call the dir `pants-donefiles` and 🚢?

Reply:
But I'm also still in favor of the files. I'm not too worried about pyenv clashes. 

But not enough to convince anyone 😅

Reply:
Well, you haven't presented an argument in favor other than it's weird to you, and I have presented at least a plausib;e argument against... I will rename to pants-donefiles though, that makes sense.

Reply:
Oh wait, you were suggesting `<PYENV_ROOT>.<specific version>.DONE`, not `<PYENV_ROOT>/versions/<specific version>.DONE` ? That seems far less risky. But still, I prefer the directory.

Reply:
Actually neither: `<PYENV_ROOT>/<specific version>.DONE`

Reply:
IIRC the root is a symlinked dir, so everything has to be under it

Reply:
The argument is that at the top level it's obvious from a glance what is done and what isn't and doesn't require any parent dirs to exist

Reply:
Ah yes, that was a confusing typo, I meant to type `<PYENV_ROOT>/<specific version>.DONE`

Reply:
I should've been more explicit in my original message.

Reply:
Well, did you want to use

Reply:
Well you would have seen that if I had pushed to the right branch! 🤦

Reply:
Can we still lay them flat so they aren't nested (so use `<version>.DONE`) and it's up to you on whether those live in the root or in a dedicated subdir?

Reply:
Done, but if I may gently suggest, for the future, that it's worth considering if these kind of tiny suggestions, based on personal preference more than technical or product considerations, are worth the time and CI resources they cost?

Reply:
To me they are, because Pants is an enormous trove of code I/we have to maintain. Making code easier to reason about makes maintenance easier. I'd go even farther and argue (and there's likely evidence to suggest) clean code has fewer bugs. Those bugs translate to much higher global resource usage than one or two CI runs in the Pants repo (especially considering that fixing the bug is _already_ one CI run :wink:)

If instead of the CI strawman you wanted to argue that it wasn't worth _your_ time and resources, my answer would be the same, but certainly have less legs :smile:

Reply:
(I also don't see the change for flattening the directories. The last push was 2 hours ago)

Reply:
Goddamn it, I keep forgetting to push to origin, I'm so used to pushing to a fork.

Reply:
> To me they are, because Pants is an enormous trove of code I/we have to maintain. Making code easier to reason about makes maintenance easier. I'd go even farther and argue (and there's likely evidence to suggest) clean code has fewer bugs. Those bugs translate to much higher global resource usage than one or two CI runs in the Pants repo (especially considering that fixing the bug is _already_ one CI run 😉)

Of course, I agree with that statement 100%. But your suggestions in this case did not, to my understanding, make the code easier to reason about. They were, AFAICT, about a fairly arbitrary preference of filesystem layout.